### ISSUE TITLE
clarify that OpVariable is decorated with Location, not OpTypeStruct

### ISSUE THREAD

**Issue by programmerjake** [KhronosGroup/Vulkan-Docs#1203]
The spec seems to imply that the `OpTypeStruct` is decorated with `Location`, however all code examples I could find have the `OpVariable` decorated instead.

https://github.com/KhronosGroup/Vulkan-Docs/blob/3fe259531ff0493ed9915bdb9ce6340bb5714230/chapters/interfaces.txt#L222-L231

**Comment by johnkslang** [comment]
When the structure is a top-level member of a block, the decoration should go on the block's member:

```
struct S { int a; };
out bname {
    layout(location = 3) S s;  // MemberDecorate bname 0 Location 3
};
```

While standalone variables should have the decoration on the variable:

```
layout(location = 3) out struct S { int a; } t;  //  Decorate t Location 3
```

We will look at making the language correctly state this.

**Comment by oddhack** [comment]
This should be fixed in the 1.2.137 spec update.
